pipeline/pipeline/resource.py

```diff
@@ -96,13 +130,63 @@ def _get_path(self, directory):
 
 
 class TaskResourceFile(ResourceFile):
+    """
+    Class representing an intermediate file from a task.
+
+    Notes
+    -----
+    All :class:`.TaskResourceFile` are temporary files and must be written
+    to a permanent location using :func:`.Pipeline.write_output` if the output needs
+    to be saved.
+    """
+
     def _get_path(self, directory):
         assert self._source is not None
         assert self._value is not None
         return directory + '/' + self._source._uid + '/' + self._value
 
 
 class ResourceGroup(Resource):
+    """
+    Class representing a mapping of identifiers to a resource file.
+
+    Examples
+    --------
+
+    Initialize a pipeline and create a new task:
+
+    >>> p = Pipeline()
+    >>> t = p.new_task()
+
+    Read a set of input files as a resource group:
+
+    >>> bfile = p.read_input_group(bed="data/example.bed",
+    ...                            bim="data/example.bim",
+    ...                            fam="data/example.fam")
+
+    Reference the entire file group:
+
+    >>> t.command(f"plink --bfile {bfile} --geno 0.2 --out {t.ofile}")
```

Comment:
How does it know what to put into `bfile` in this scenario?

Reply:
`bfile` is a Resource Group created with `read_resource_group`. It creates a uid for the file root and then the extensions are based on the keyword argument names to `read_resource_group`.